L1Trigger/TrackFindingTracklet/src/PurgeDuplicate.cc

```diff
@@ -108,264 +108,327 @@ void PurgeDuplicate::execute(std::vector<Track>& outputtracks_, unsigned int iSe
-          for (unsigned int stcount = 0; stcount < stubsTrk1.size(); stcount++) {
-            int i = stubsTrk1[stcount].first;
-            int reg = (i > 0 && i < 10) * (i - 1) + (i > 10) * (i - 5) - (i < 0) * i;
-            double nres = getPhiRes(inputtracklets_[itrk], fullStubslistsTrk1[stcount]);
-            double ores = 0;
-            if (URStubidsTrk1[reg] != -1)
-              ores = getPhiRes(inputtracklets_[itrk], fullStubslistsTrk1[URStubidsTrk1[reg]]);
-            if (URStubidsTrk1[reg] == -1 || nres < ores) {
-              URStubidsTrk1[reg] = stcount;
+          } else if (settings_.mergeComparison() == "CompareBest") {
+            std::vector<const Stub*> fullStubslistsTrk1 = inputstublists_[itrk];
+            std::vector<const Stub*> fullStubslistsTrk2 = inputstublists_[jtrk];
+
+            // Arrays to store the index of the best stub in each region
+            int URStubidsTrk1[16];
+            int URStubidsTrk2[16];
+            for (int i = 0; i < 16; i++) {
+              URStubidsTrk1[i] = -1;
+              URStubidsTrk2[i] = -1;
             }
-          }
-          // For each stub on the second track, find the stub with the best residual and store its index in the URStubidsTrk1 array
-          for (unsigned int stcount = 0; stcount < stubsTrk2.size(); stcount++) {
-            int i = stubsTrk2[stcount].first;
-            int reg = (i > 0 && i < 10) * (i - 1) + (i > 10) * (i - 5) - (i < 0) * i;
-            double nres = getPhiRes(inputtracklets_[jtrk], fullStubslistsTrk2[stcount]);
-            double ores = 0;
-            if (URStubidsTrk2[reg] != -1)
-              ores = getPhiRes(inputtracklets_[jtrk], fullStubslistsTrk2[URStubidsTrk2[reg]]);
-            if (URStubidsTrk2[reg] == -1 || nres < ores) {
-              URStubidsTrk2[reg] = stcount;
+            // For each stub on the first track, find the stub with the best residual and store its index in the URStubidsTrk1 array
+            for (unsigned int stcount = 0; stcount < stubsTrk1.size(); stcount++) {
+              int i = stubsTrk1[stcount].first;
+              int reg = (i > 0 && i < 10) * (i - 1) + (i > 10) * (i - 5) - (i < 0) * i;
+              double nres = getPhiRes(inputtracklets_[itrk], fullStubslistsTrk1[stcount]);
+              double ores = 0;
+              if (URStubidsTrk1[reg] != -1)
+                ores = getPhiRes(inputtracklets_[itrk], fullStubslistsTrk1[URStubidsTrk1[reg]]);
+              if (URStubidsTrk1[reg] == -1 || nres < ores) {
+                URStubidsTrk1[reg] = stcount;
+              }
+            }
+            // For each stub on the second track, find the stub with the best residual and store its index in the URStubidsTrk1 array
+            for (unsigned int stcount = 0; stcount < stubsTrk2.size(); stcount++) {
+              int i = stubsTrk2[stcount].first;
+              int reg = (i > 0 && i < 10) * (i - 1) + (i > 10) * (i - 5) - (i < 0) * i;
```

Comment:
Ugh! This line needs an explanatory comment. And unnamed constants like 5 or 10 are not allowed.

Reply:
I didn't write this line, so I don't know what the 5 or 10 refer to. I can take a closer look and see if I can define constants for them.